maven-embedder/src/main/java/org/apache/maven/cli/event/ExecutionEventLogger.java

```diff
@@ -143,7 +143,11 @@ public void sessionStarted( ExecutionEvent event )
                         project.getName(), chars( ' ', ( len > 0 ) ? len : 1 ), project.getPackaging() );
             }
 
-            totalProjects = projects.size();
+            final List<MavenProject> allProjects = event.getSession().getAllProjects();
```

Comment:
Perhaps it might be worth noting that the `MavenSession#getAllProjects()` [states](https://github.com/apache/maven/blob/3c487fa697c3ff29d6c79ab5cf6e834abfd143b3/maven-core/src/main/java/org/apache/maven/execution/MavenSession.java#L273) `This is a provisional method and may be removed`. However long that may have been there :see_no_evil:.

Reply:
Yeah that's a good point. It's been there since 2014, though and the documentation of the member field shows exactly what we need :)

```
/**
     * The full set of projects before any potential constraining by --projects. Useful in the case where you want to
     * build a smaller set of projects but perform other operations in the context of your reactor.
     */
```

I'm inclined to keep it, what do you think? :)

Reply:
As hard as it sounds, I deprecation without even a slight note is quite useless...

Reply:
It makes sense to me to keep using this method. Perhaps the comment could be removed to avoid confusion as it is not annotated with `@Deprecated` and it is used in multiple places. I agree that these kind of "deprecation messages" are quite useless indeed.

Reply:
Makes sense, but I am not sure what the policy is on that. @michael-o Should I remove it as suggested, what do you think?

Reply:
I would at least raise a JIRA issue.

Reply:
A JIRA for removing that method and adding the JIRA number into the TODO?

Reply:
Yes